src/components/Molecules/Membership/Form/Form.js

```diff
@@ -0,0 +1,166 @@
+
+import Text from '../../../Atoms/Text/Text';
+import MoneyBuy from '../MoneyBuy/MoneyBuy';
+import {
+  onKeyPress,
+  isAmountValid,
+  getUrlParameter,
+  isInputMatchBoxValue
+} from '../../../../utils/Membership';
+import {
+  Button,
+  Copy,
+  Error,
+  FormFieldset,
+  FormWrapper,
+  Label,
+  Form,
+  MoneyBuys,
+  AmountField
+} from '../Membership.style';
+
+const Signup = ({ data: { regularGiving }, ...rest }) => {
+  // It's used to hightlight one of the money buy box when page load
+  const [isSelected, setIsSelected] = useState(true);
+  const [moneyBoxes, setMoneyBoxes] = useState({
+    box1: '',
+    box2: '',
+    box3: ''
+  });
+
+  const [errorMsg, setErrorMsg] = useState(false);
+  const [amountDonate, setAmountDonate] = useState('');
+  const [userInput, setUserInput] = useState('');
+  const [boxBorderColor, setBoxBorderColor] = useState('');
+  const [inputBorderColor, setInputBorderColor] = useState(false);
+  const [moneyBuyCopy, setMoneyBuyCopy] = useState(true);
+
+  const selectMoneyBuy = (copy, value) => {
+    setIsSelected(false);
+    if (errorMsg) setErrorMsg(false);
+    const isUserInputMatch = userInput * 1 === value * 1;
+
+    if (inputBorderColor) {
+      if (!isUserInputMatch) {
+        setInputBorderColor(false);
+        setUserInput('');
+      }
+    }
+    setBoxBorderColor(value);
+    setMoneyBuyCopy(copy);
+    setAmountDonate(value);
+  };
+
+  // Handle user input
+  const handleChange = (input, otherDescription) => {
+    if (!isAmountValid(input)) {
+      if (moneyBuyCopy) setMoneyBuyCopy(false);
+      if (!errorMsg) setErrorMsg(true);
+    } else {
+      setBoxBorderColor(false);
+      if (errorMsg) setErrorMsg(false);
+      setMoneyBuyCopy(otherDescription);
+      setAmountDonate(input);
+    }
+    // Highlight both input and box
+    isInputMatchBoxValue(moneyBoxes, selectMoneyBuy, setAmountDonate, input);
+    setInputBorderColor(true);
+    setUserInput(input);
+  };
+
+  const hightlightInput = (value, otherDescription) => {
+    if (isSelected) setIsSelected(false);
+    if (errorMsg) {
+      setMoneyBuyCopy(false);
+    } else if (!value) {
+      setAmountDonate(0);
+      setBoxBorderColor(false);
+      setMoneyBuyCopy(otherDescription);
+    }
+    setInputBorderColor(true);
+  };
+
+  useEffect(() => {
+    regularGiving.moneybuys.map((moneyBuy, index) => {
+      const box = `box${index + 1}`;
+      // eslint-disable-next-line no-shadow
+      setMoneyBoxes(moneyBoxes => ({ ...moneyBoxes, [box]: moneyBuy }));
+      return (
+        isSelected &&
+        index === 1 &&
+        (setMoneyBuyCopy(moneyBuy.description), setAmountDonate(moneyBuy.value))
+      );
+    });
+  }, [isSelected, regularGiving.moneybuys]);
+
+  const boxes = regularGiving.moneybuys.map(({ value, description }, index) => (
+    <MoneyBuy
+      isSelected={index === 1 && isSelected}
+      boxBorderColor={boxBorderColor}
+      current={value}
+      amount={`${value}`}
```

Comment:
`{value}` same thing?

Reply:
Props validation was set to string will set it to number